Checkout: Remove SecurePaymentForm

sirbrillig opened the pull request:
#### Changes proposed in this Pull Request

This removes the `SecurePaymentForm` component from "old" checkout, which is no longer in use and many of its otherwise unused dependencies. This will allow us to eventually remove the `TransactionFlow` object which is the last major consumer of the deprecated `CartStore`.

#### Testing instructions

Verify that these components and functions are not being used anywhere. Just compiling Calypso should be sufficient for the files that were fully removed. 

The single functions that were removed should be verified as unused individually:

- isPaidForFullyInCredits
- isAntiSpamConflicting
- isProductConflictingWithSite
- getCheckoutIncompatibleProducts

And the type:

- IncompatibleProducts

sirbrillig committed:
Remove SecurePaymentForm

sirbrillig committed:
Remove FreeTrialConfirmationBox

sirbrillig committed:
Remove IncompatibleProductMessage

sirbrillig committed:
Remove unused functions in conflicts.js

sirbrillig committed:
Remove FreeCartPaymentBox

sirbrillig committed:
Remove PayButton

sirbrillig committed:
Remove SubscriptionText

sirbrillig committed:
Remove isPaidForFullyInCredits

sirbrillig committed:
Remove conflicts tests

sirbrillig committed:
Remove CartToggle

sirbrillig committed:
Remove toggleCartOnMobile

sirbrillig committed:
Remove SecondaryCart

sirbrillig committed:
Remove non-children rendering for Checkout

It's only used currently by CheckoutContainer and that always passes children.

sirbrillig committed:
Remove CheckoutData

It was only used when CheckoutContainer had clearTransaction={ false },
which is never the case.

nbloomf commented:
The single function names no longer appear in the source tree, but I think there's still a usage of `IncompatibleProducts`:

```
client/my-sites/checkout/checkout/incompatible-product-notice.tsx
15:import type { IncompatibleProducts } from 'calypso/state/sites/products/conflicts';
18:	incompatibleProducts: IncompatibleProducts | null;
```

nbloomf commented:
This is great! I clicked around in checkout just to make sure nothing bad happened. I did leave one comment above, but once that is resolved I think this is good to go.

sirbrillig committed:
Remove IncompatibleProductNotice

sirbrillig commented:
Good find! I don't know how I missed that. Probably just because there's so much being removed. Thanks! That component also looks unused so I've removed it.

pottedmeat commented on client/components/data/checkout/index.jsx:
```diff
@@ -1,33 +0,0 @@
- */
-import StoreConnection from 'calypso/components/data/store-connection';
-import CartStore from 'calypso/lib/cart/store';
-import TransactionStore from 'calypso/lib/transaction/store';
```
It looks like this is the only time this import was used. Would anything else use this?

sirbrillig replied:
Wow, you're right. And not only that, but it looks like the entirety of that directory might be unused. I will verify and then trash it...

pottedmeat commented on client/my-sites/checkout/cart/secondary-cart.jsx:
```diff
@@ -1,109 +0,0 @@
-import CartBody from './cart-body';
-import CartFreeUserPlanUpsell from 'calypso/my-sites/checkout/cart/cart-free-user-plan-upsell';
-import CartMessages from './cart-messages';
-import CartSummaryBar from './cart-summary-bar';
```
import doesn't seem to be used elsewhere

sirbrillig replied:
You're right. There's another one.

sirbrillig committed:
Remove more unused code in old Checkout

sirbrillig committed:
Remove DomainDetailsForm

sirbrillig committed:
Remove CreditCardSelector

sirbrillig committed:
Remove SecurePaymentFormPlaceholder

sirbrillig committed:
Copy myFormatCurrency into composite checkout

We will be removing the otherwise unused subscription-length-picker
directory.

sirbrillig committed:
Remove SubscriptionLengthPicker from devdocs

sirbrillig committed:
Remove SubscriptionLengthPicker

sirbrillig committed:
Remove PendingPaymentBlocker

sirbrillig committed:
Remove PaymentBox

sirbrillig committed:
Remove detectWebPaymentMethod/getWebPaymentMethodName

sirbrillig committed:
Remove getExitCheckoutUrl

sirbrillig committed:
Remove handleDispatch in CartSynchronizer

TRANSACTION_STEP_SET no longer exists

sirbrillig committed:
Remove NewCardForm

sirbrillig committed:
Remove transaction events from CartStore

The transaction store is no longer used

sirbrillig committed:
Remove lib/transaction

sirbrillig committed:
Remove CartSummaryBar

sirbrillig commented:
With the additional removals (whew! that was a lot) we should double-check a few things like the general new checkout flow including the variant picker and the domain contact form since, in addition to the "add credit card" form as this comes close to those things. Finally we should test the one-click upsell described in https://github.com/Automattic/wp-calypso/pull/46594 which still uses pieces of old checkout like `handleCheckoutCompleteRedirect`.

sirbrillig committed:
Remove tests that reference TransactionStore

sirbrillig committed:
Remove unused dispatchToken in CartSynchronizer

sirbrillig commented:
Failing e2e tests should be fixed by https://github.com/Automattic/wp-calypso/pull/48099

pottedmeat commented on client/blocks/subscription-length-picker/index.jsx:
```diff
@@ -1,134 +0,0 @@
-		);
-
-		return (
-			<div className="subscription-length-picker">
```
Looks like these classes are still around in style sheets

pottedmeat commented on client/lib/cart/store/index.js:
```diff
@@ -21,14 +21,10 @@ import {
 } from 'calypso/lib/cart/action-types';
-import {
-	TRANSACTION_NEW_CREDIT_CARD_DETAILS_SET,
-	TRANSACTION_PAYMENT_SET,
```
These are deleted but appear as strings in lib/redux-bridge. Seems related.

sirbrillig replied:
Yes... I think those actions are copying the country code into the redux store for some purpose. That should also all probably be removed, but it's a bit tangled and I don't want to dive into that just yet.

pottedmeat commented on client/lib/cart/store/index.js:
```diff
@@ -21,14 +21,10 @@ import {
 import emitter from 'calypso/lib/mixins/emitter';
 import cartSynchronizer from './cart-synchronizer';
 import PollerPool from 'calypso/lib/data-poller';
-import { recordEvents, recordUnrecognizedPaymentMethod } from 'calypso/lib/analytics/cart';
```
I don't think `recordUnrecognizedPaymentMethod` is used anywhere else.

pottedmeat commented on client/lib/cart/store/index.js:
```diff
@@ -38,7 +34,6 @@ import {
 	fillInAllCartItemAttributes,
 	setTaxCountryCode,
 	setTaxPostalCode,
-	setTaxLocation,
```
I don't think `setTaxLocation` is used anywhere else.

sirbrillig replied:
I think we might as well leave this for now as it's part of the old CartStore which will be deleted wholesale at some point soon 🤞

sirbrillig replied:
Nevermind, I removed it anyway.

pottedmeat commented on client/lib/cart/store/test/index.js:
```diff
@@ -2,13 +2,6 @@
- * Internal dependencies
- */
-import { transactionPaymentSetActions, paymentActionLocations } from './fixtures/actions';
-import { recordUnrecognizedPaymentMethod } from 'calypso/lib/analytics/cart';
```
It looks like this is the only place this was used.

pottedmeat commented on client/lib/cart/store/test/index.js:
```diff
@@ -2,13 +2,6 @@
- */
-import { transactionPaymentSetActions, paymentActionLocations } from './fixtures/actions';
-import { recordUnrecognizedPaymentMethod } from 'calypso/lib/analytics/cart';
-import { setTaxLocation } from 'calypso/lib/cart-values';
```
It looks like this is the only place this was used.

pottedmeat commented on client/lib/checkout/index.js:
```diff
@@ -6,23 +6,13 @@ import { parse } from 'url'; // eslint-disable-line no-restricted-imports
-	hasRenewalItem,
-	getRenewalItems,
-	hasDomainRegistration,
-	hasDomainMapping,
```
It looks like this is the only place this was used.

sirbrillig replied:
Good eye, however I think I'm going to leave this around for now as it seems like it might have a utility value later on.

pottedmeat commented on client/my-sites/checkout/cart/secondary-cart.jsx:
```diff
@@ -1,109 +0,0 @@
-	render() {
-		const { cart, selectedSite, isJetpackNotAtomic, isShowingOnMobile } = this.props;
-
-		const cartClasses = classNames( 'secondary-cart', {
```
It looks like there are still some of these style rules remaining.

pottedmeat commented on client/my-sites/checkout/checkout/domain-details-form.jsx:
```diff
@@ -1,329 +0,0 @@
- * Internal dependencies
- */
-import QueryContactDetailsCache from 'calypso/components/data/query-contact-details-cache';
-import QueryTldValidationSchemas from 'calypso/components/data/query-tld-validation-schemas';
```
It doesn't look like this is used anywhere else.

sirbrillig replied:
I'm going to leave that for now because it seems like it could be important if we want to add more client-side validation in the future.

pottedmeat reviewed (changes requested):
There's a lot of stuff that can be pruned which isn't super important but I think we need to make sure to clean up a lot of the style rules that are unused now.

sirbrillig committed:
Remove some old checkout CSS

Probably more of this can be removed but we need to be careful since
some styles for components pulled in from old checkout may be merely
being overridden in new checkout.

sirbrillig committed:
Remove recordUnrecognizedPaymentMethod

sirbrillig committed:
Remove setTaxLocation

sirbrillig committed:
Remove secondary-cart styles

sirbrillig commented:
Thanks! I did end up removing a lot of what you suggested and left comments for the rest. I also rebased this which hopefully should allow the tests to pass; we'll see.

tyxla commented:
By getting rid of the legacy transactions store, we also have one less store to migrate to Redux - see https://github.com/Automattic/wp-calypso/issues/24217

Thanks @sirbrillig 🙌
